yjit/src/asm/x86_64/mod.rs

```diff
@@ -117,90 +133,140 @@ pub enum X86Opnd
     IPRel(i32)
 }
 
+impl X86Opnd {
+    fn rex_needed(&self) -> bool {
+        match self {
+            X86Opnd::None => unreachable!(),
+            X86Opnd::Imm(_) => false,
+            X86Opnd::Reg(reg) => (reg.reg_no > 7 || (reg.num_bits == 8 && reg.reg_no >= 4 && reg.reg_no <= 7)),
+            X86Opnd::Mem(mem) => (mem.base_reg_no > 7 || (mem.idx_reg_no.unwrap_or(0) > 7)),
+            X86Opnd::IPRel(_) => false
+        }
+    }
+
+    fn sib_needed(&self) -> bool {
+        match self {
+            X86Opnd::Mem(mem) => {
+                mem.idx_reg_no.is_some() ||
+                mem.base_reg_no == RSP_REG_NO ||
+                mem.base_reg_no == R12_REG_NO
+            },
+            _ => false
+        }
+    }
```

Comment:
Putting this method here makes sense and I like it, but it would have been nice to preserve the original comment explaining the purpose of the method, even if it seems obvious.

Reply:
👍